pageserver/src/tenant/mgr.rs

```diff
@@ -1365,35 +1367,12 @@ impl TenantManager {
         }
     }
 
-    pub(crate) async fn delete_tenant(
+    pub(crate) async fn delete_tenant_attached(
         &self,
-        tenant_shard_id: TenantShardId,
+        slot_guard: SlotGuard,
+        tenant: Arc<Tenant>,
         activation_timeout: Duration,
-    ) -> Result<(), DeleteTenantError> {
-        super::span::debug_assert_current_span_has_tenant_id();
```

Comment:
Why remove this?

Reply:
Ah, it moved to the caller... Is this still supposed to be `pub(crate)` then?

Reply:
Yes, this can be private: updated in  6ebbfd0d687fcd140312b3a0a4f8d890be0c9c47